.cbmc-batch/array_list_proofs.c

```diff
@@ -0,0 +1,75 @@
+#include <aws/common/array_list.h>
+#include <assert.h>
+#include <stddef.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <proof_helpers.h>
+
+#define MAX_INITIAL_ITEM_ALLOCATION 2
+#define MAX_ITEM_SIZE 15
+
+void aws_array_list_init_dynamic_verify(void) {
+    struct aws_array_list *list;
+    ASSUME_VALID_MEMORY(list);
+    struct aws_allocator *allocator;
+    ASSUME_DEFAULT_ALLOCATOR(allocator);
+    size_t item_count = nondet_size_t();
+    size_t item_size = nondet_size_t();
+
+    aws_array_list_init_dynamic(list, allocator, item_count, item_size);
+    
+    // some guarantees
```

Comment:
we avoid c++ style comments. Let's be consistent in this portion of the code-base too.